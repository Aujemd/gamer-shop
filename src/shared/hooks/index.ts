export { useStoredCart } from "./use-stored-cart";
