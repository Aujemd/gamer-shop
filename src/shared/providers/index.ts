export * from "./data-fetching-provider";
