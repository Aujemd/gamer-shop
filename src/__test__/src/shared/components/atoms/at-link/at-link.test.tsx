import { fireEvent, render, screen } from "@testing-library/react";

import { AtLink } from "@/shared";

describe("atLink Component", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<AtLink href="/home">Home</AtLink>);
        expect(asFragment()).toMatchSnapshot();
    });
    it("renders a link with the correct href", () => {
        render(
            <AtLink
                href="/test"
                data-testid="at-link-test-id"
            >
                test
            </AtLink>,
        );
        const link = screen.getByTestId("at-link-test-id");
        expect(link).toBeInTheDocument();
        expect(link).toHaveAttribute("href", "/test");
    });

    it("renders children correctly", () => {
        render(<AtLink href="/test">Home</AtLink>);
        expect(screen.getByText("Home")).toBeInTheDocument();
    });

    it("applies custom className", () => {
        render(
            <AtLink
                href="/test"
                className="custom-class"
                data-testid="at-link-test-id"
            >
                Test
            </AtLink>,
        );
        expect(screen.getByTestId("at-link-test-id")).toHaveClass("custom-class");
    });

    it("passes additional props to the anchor", () => {
        render(<AtLink href="/props" data-testid="at-link-test-id">Props</AtLink>);
        expect(screen.getByTestId("at-link-test-id")).toBeInTheDocument();
    });

    it("calls onClick handler when clicked", async () => {
        const handleClick = jest.fn();
        render(<AtLink href="/click" onClick={handleClick} data-testid="at-link-test-id">test</AtLink>);
        fireEvent.click(screen.getByTestId("at-link-test-id"));
        expect(handleClick).toHaveBeenCalled();
    });
});
