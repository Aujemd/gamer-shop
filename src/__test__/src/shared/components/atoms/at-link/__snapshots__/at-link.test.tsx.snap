// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atLink Component should render correctly 1`] = `
<DocumentFragment>
  <a
    class="font-archivo font-medium text-base leading-4 tracking-normal "
    href="/home"
  >
    Home
  </a>
</DocumentFragment>
`;
